merlin/models/tf/models/benchmark.py

```diff
@@ -94,3 +98,107 @@ def NCFModel(
     model = Model(ncf, prediction_tasks)
 
     return model
+
+
+def WideAndDeepModel(
+    schema: Schema,
+    embedding_dim: int,
+    wide_schema: Optional[Schema] = None,
+    deep_schema: Optional[Schema] = None,
+    deep_block: Optional[Block] = None,
+    deep_input_block: Optional[Block] = None,
+    wide_input_block: Optional[Block] = None,
+    prediction_tasks: Optional[
+        Union[PredictionTask, List[PredictionTask], ParallelPredictionBlock]
+    ] = None,
+    embedding_option_kwargs: dict = {},
+    **kwargs,
+) -> Model:
+    """Wide-and-Deep-model architecture.
+
+    Example Usage::
+        wide_deep = WideAndDeepModel(schema, embedding_dim=32)
+        wide_deep.compile(optimizer="adam")
+        wide_deep.fit(train_data, epochs=10)
+
+    References
+    ----------
+    [1] Cheng, Koc, Harmsen, Shaked, Chandra, Aradhye, Anderson et al. "Wide & deep learning for
+    recommender systems." In Proceedings of the 1st workshop on deep learning for recommender
+    systems, pp. 7-10. (2016).
+
+    Parameters
+    ----------
+    schema : Schema
+        The `Schema` with the input features
+    embedding_dim : int
+        Dimension of the embeddings
+    wide_schema : optional
+        The 'Schema' of input features for wide model, by default all features would be sent
+        to wide model, if specified, only features in wide_schema would be sent to wide model
+    deep_schema : optional
+        The 'Schema' of input features for deep model, by default all features would be sent to
+        deep model. deep_schema and wide_schema could contain the same features
+    prediction_tasks: optional
+        The prediction tasks to be used, by default this will be inferred from the Schema.
+    embedding_option_kwargs: Optional[dict]
+        Additional arguments to provide to `EmbeddingOptions` object for embeddings tables setting.
+        Defaults to {}
+
+    Returns
+    -------
+    Model
+
+    """
+
+    prediction_tasks = parse_prediction_tasks(schema, prediction_tasks)
+
+    if schema is None:
+        raise ValueError("The schema is required by Wide and Deep Model")
+
+    if embedding_dim is None:
+        raise ValueError("The embedding_dim is required")
+
+    if not wide_schema:
+        warnings.warn("If not specify wide_schema, all features would be sent to wide " "model")
+        wide_schema = schema
+
+    if not deep_schema:
+        deep_schema = schema
```

Comment:
Would it make sense to allow users to define and train only the Wide model?

Reply:
I think if users only want to train a wide model, they would simply choose a dense layer or a linear layer, no need to use wide and deep model.

Reply:
I think feature interactions should be done in the model side (and not in preprocessing side), for being more flexible for a Data Scientist to experiment with different feature combinations without the need of preprocessing a new version of the dataset.  
I agree that the feature interactions of the wide part of W&D could be defined by using a `CrossFeature` block, so that it is reusable by other models which include a linear model with feature interaction.
That was the design choice of TF ([tf.feature_column.crossed_column](https://www.tensorflow.org/api_docs/python/tf/feature_column/crossed_column)).
I know Sara's PR #301 has a prototype of a `CrossFeatures` block which leverages `tf.feature_column.crossed_column` that can be useful.

Reply:
Created [HashedCross ](https://github.com/NVIDIA-Merlin/models/pull/587)and [HashedCrossAll](https://github.com/NVIDIA-Merlin/models/pull/590)